worker/main.go

```diff
@@ -153,6 +213,94 @@ func CreateSlide(ctx worker.Context, args ...interface{}) error {
 	_, err = keycloak.GetUser(userEmail)
 	handleError(err)
 	fmt.Println("Created Slide for: ", userEmail)
+	err = saveFileToDB(userEmail, outputFile)
+	handleError(err)
+	return err
+}
+
+// createSurveyInput creates a new survey input file for the user
+func createSurveyInput(surveyID string, tx *sqlx.Tx) error {
+
+	surveyData := []SurveyData{}
+
+	err := tx.Select(&surveyData, `WITH current_survey AS(
+		SELECT mappa.ipsative_responses.answer_id, mappa.survey_responses.user_id,
+			mappa.ipsative_responses.points_assigned
+		FROM mappa.ipsative_responses
+		INNER JOIN mappa.survey_responses
+		ON mappa.ipsative_responses.user_id=mappa.survey_responses.user_id
+		WHERE mappa.ipsative_responses.survey_response_id = $1)
+		SELECT mappa.ipsative_answers.uuid, (
+			SELECT mappa.ipsative_questions.order_number
+			FROM mappa.ipsative_questions
+			WHERE mappa.ipsative_answers.question_id =  mappa.ipsative_questions.uuid)
+			AS question_order, mappa.ipsative_answers.order_number AS answer_order, (
+				SELECT mappa.ipsative_categories.category
+				FROM mappa.ipsative_categories
+				WHERE mappa.ipsative_answers.category_id =  mappa.ipsative_categories.uuid)
+			, points_assigned
+		FROM current_survey
+		INNER JOIN mappa.ipsative_answers
+		ON mappa.ipsative_answers.uuid = current_survey.answer_id`, surveyID)
+	handleError(err)
+	fileContents := "question,Process,Compliance,Autonomy,Trust,respondent\n"
+	questions := make([][]int, 10)
+	// Collect answer point assignments
+	for i := range surveyData {
+		questions[surveyData[i].Question-1] = append(
+			questions[surveyData[i].Question-1],
+			surveyData[i].Points)
+	}
+	// Supports only 1 respondent currently.
+	respondent := "1"
+	for i := range questions {
+		fileContents += fmt.Sprintf("%d,%s,%s\n", i, strings.Trim(strings.Join(strings.Fields(fmt.Sprint(questions[i])), ","), "[]"), respondent)
+	}
+
+	// Create new file
+	file, err := os.Create("/tmp/" + surveyID + ".csv")
```

Comment:
I haven't fully thought through the scenario, but I think using predictable filenames in /tmp has some security risks: https://www.owasp.org/index.php/Insecure_Temporary_File

I also noticed that it doesn't seem like the file is removed after being used.

What do you think about using something like https://golang.org/pkg/io/ioutil/#TempFile and deleting the temporary file(s) once the job completes?

Reply:
@statik I have refactored how these temp files are handled. It included moving some things around but let me know what you think. Thanks!